homeassistant/helpers/entity_component.py

```diff
@@ -176,14 +179,14 @@ async def async_unload_entry(self, config_entry: ConfigEntry) -> bool:
 
     async def async_extract_from_service(
         self, service_call: ServiceCall, expand_group: bool = True
-    ) -> list[entity.Entity]:
+    ) -> list[_EntityT]:
         """Extract all known and available entities from a service call.
 
         Will return an empty list if entities specified but unknown.
 
         This method must be run in the event loop.
         """
-        return await service.async_extract_entities(
+        return await service.async_extract_entities(  # type: ignore[return-value]
```

Comment:
If you use a TypeVar in `async_extract_entities` too, the `type: ignore` shouldn't be necessary.
https://github.com/home-assistant/core/blob/fd05d949cc76d340dad9bf7dc3ff5d937d8d1b1e/homeassistant/helpers/service.py#L276-L286

Reply:
Opened #78490